File: dotnet/Liberator.Driver/Performance/ITimer.cs
using System;

namespace Liberator.Driver.Performance
{
    /// <summary>
    /// Root interface for timer implementations
    /// </summary>
    public interface ITimer
    {
        /// <summary>
        /// Start time of the timer
        /// </summary>
        DateTime StartTime { get; set; }

        /// <summary>
        /// End time for the timer
        /// </summary>
        DateTime EndTime { get; set; }

        /// <summary>
        /// Duration for the timer instance
        /// </summary>
        TimeSpan Duration { get; set; }

        /// <summary>
        /// Start the timer
        /// </summary>
        void Start();

        /// <summary>
        /// Stop the timer
        /// </summary>
        void Stop();
    }
}

File: java/src/main/java/org/liberator/ratdriver/performance/ITimer.java
package org.liberator.ratdriver.performance;

import lombok.Getter;
import lombok.Setter;

import java.time.Instant;
import java.util.Date;

public interface ITimer {
    /**
     * Start the timer
     */
    void Start();

    /**
     * Stop the timer
     */
    void Stop();
}
